packages/battery/android/src/main/java/io/flutter/plugins/battery/BatteryPlugin.java

```diff
@@ -36,18 +39,32 @@ public static void registerWith(PluginRegistry.Registrar registrar) {
   }
 
   BatteryPlugin(PluginRegistry.Registrar registrar) {
-    this.registrar = registrar;
+    this.application = ((Application) registrar.context());
+    application.registerActivityLifecycleCallbacks(lifecycleCallbacks);
   }
 
-  private final PluginRegistry.Registrar registrar;
+  private final Application application;
+  private final IntentFilter chargedFilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
   private BroadcastReceiver chargingStateChangeReceiver;
 
+  private final ActivityLifecycleCallbacks lifecycleCallbacks =
+      new EmptyActivityLifecycleCallbacks() {
+        @Override
+        public void onActivityDestroyed(Activity activity) {
+          application.unregisterActivityLifecycleCallbacks(lifecycleCallbacks);
```

Comment:
So we should register this callback only if the current Activity is the flutter Activity too, right?

Reply:
Since the plugin should only be registered from an Activity that is attached to an engine, it shouldn't be an issue.
See how we're dealing with the same issue in another plugin: https://github.com/flutter/plugins/blob/a2b5a777beaf54df591303c8e7c545c8950dcf9b/packages/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapsPlugin.java#L88